stl/inc/algorithm

```diff
@@ -3924,6 +4136,73 @@ _CONSTEXPR20 void make_heap(_RanIt _First, _RanIt _Last) { // make [_First, _Las
     _STD make_heap(_First, _Last, less<>());
 }
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // VARIABLE ranges::make_heap
+    class _Make_heap_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <random_access_iterator _It, sentinel_for<_It> _Se, class _Pr = ranges::less, class _Pj = identity>
+            requires sortable<_It, _Pr, _Pj>
+        constexpr _It operator()(_It _First, _Se _Last, _Pr _Pred = {}, _Pj _Proj = {}) const {
+            _Adl_verify_range(_First, _Last);
+            auto _UFirst = _Get_unwrapped(_STD move(_First));
+            auto _ULast  = _Get_unwrapped(_STD move(_Last));
+            if constexpr (is_same_v<_It, _Se>) {
+                _Seek_wrapped(_First, _ULast);
+                _Make_heap_unchecked(_STD move(_UFirst), _STD move(_ULast), _Pass_fn(_Pred), _Pass_fn(_Proj));
+            } else if constexpr (sized_sentinel_for<_Se, _It>) {
+                auto _UFinal = _UFirst + (_ULast - _UFirst);
+                _Seek_wrapped(_First, _UFinal);
+                _Make_heap_unchecked(_STD move(_UFirst), _STD move(_UFinal), _Pass_fn(_Pred), _Pass_fn(_Proj));
+            } else {
+                auto _UFinal = _RANGES next(_UFirst, _STD move(_ULast));
+                _Seek_wrapped(_First, _UFinal);
+                _Make_heap_unchecked(_STD move(_UFirst), _STD move(_UFinal), _Pass_fn(_Pred), _Pass_fn(_Proj));
+            }
+
+            return _First;
+        }
+
+        template <random_access_range _Rng, class _Pr = ranges::less, class _Pj = identity>
+            requires sortable<iterator_t<_Rng>, _Pr, _Pj>
+        constexpr borrowed_iterator_t<_Rng> operator()(_Rng&& _Range, _Pr _Pred = {}, _Pj _Proj = {}) const {
+            if constexpr (common_range<_Rng>) {
+                _Make_heap_unchecked(_Ubegin(_Range), _Uend(_Range), _Pass_fn(_Pred), _Pass_fn(_Proj));
+                return _RANGES end(_Range);
+            } else {
+                auto _ULast = _Get_final_iterator_unwrapped(_Range);
+                _Make_heap_unchecked(_Ubegin(_Range), _ULast, _Pass_fn(_Pred), _Pass_fn(_Proj));
+                auto _First = _RANGES begin(_Range);

```

Comment:
Similar to above would a `_Seek_end` function make sense that abstracts those away?

Reply:
To order my thoughts a bit. If we had a function `_Seek_end` or whatever you want to name it we could simplify the whole block to
```
    auto _ULast = _Get_final_iterator_unwrapped(_Range);
    _Make_heap_unchecked(_Ubegin(_Range), _ULast, _Pass_fn(_Pred), _Pass_fn(_Proj));
    return _Seek_end(_RANGES begin(_Range), _STD move(_ULast));
```

Reply:
Added `_Rewrap_iterator`, which accepts a range and unwrapped iterator and returns a wrapped iterator.